tensorflow/examples/tutorials/mnist/mnist_rnn.py

```diff
@@ -0,0 +1,120 @@
+from tensorflow.contrib import rnn
+
+# Import MNIST data
+from tensorflow.examples.tutorials.mnist import input_data
+mnist = input_data.read_data_sets("/tmp/mnist_data/", one_hot=True)
+
+'''
+To classify images using a recurrent neural network, we consider every image
+row as a sequence of pixels. Because MNIST image shape is 28*28px, we will then
+handle 28 sequences of 28 steps for every sample.
+'''
+
+# Training Parameters
+learning_rate = 0.001
+training_steps = 10000
+batch_size = 128
+display_step = 200
+
+# Network Parameters
+num_input = 28 # MNIST data input (img shape: 28*28)
+num_step = 28 # step number
+num_hidden_units = 128 # hidden layer num of features
+num_classes = 10 # MNIST total classes (0-9 digits)
+
+# tf Graph input
+X = tf.placeholder("float", [None, num_step, num_input])
+Y = tf.placeholder("float", [None, num_classes])
+
+# Define weights
+weights = {
+    # (28, 128)
+    'in': tf.Variable(tf.random_normal([num_input, num_hidden_units])),
+    # (128, 10)
+    'out': tf.Variable(tf.random_normal([num_hidden_units, num_classes]))
+}
+biases = {
+    # (128, )
+    'in': tf.Variable(tf.constant(0.1, shape=[num_hidden_units, ])),
+    # (10, )
+    'out': tf.Variable(tf.constant(0.1, shape=[num_classes, ]))
+}
+
+
+def RNN(x, weights, biases):
+
+    # Prepare data shape to match `rnn` function requirements
+    # Current data input shape: (batch_size, num_step, n_input)
+    # Required shape: 'num_step' tensors list of shape (batch_size, n_input)
+    x = tf.reshape(x, [-1, num_input])
+
+    # Hidden layer ==> (128 batch, 28 steps, 128 hidden)
+    x_in = tf.matmul(x, weights['in']) + biases['in']
+    x_in = tf.reshape(x_in, [-1, num_step, num_hidden_units])
+
+    # Define a lstm cell with tensorflow
+    lstm_cell = rnn.BasicLSTMCell(num_hidden_units, forget_bias=1.0, state_is_tuple=True)
+
+    # init zero state, lstm cell is consist of twp parts: (c_state, h_state)
+    init_state = lstm_cell.zero_state(batch_size, dtype=tf.float32)
+
+    # Get lstm cell output
+    outputs, states = tf.nn.dynamic_rnn(lstm_cell, x_in, initial_state=init_state, time_major=False, dtype=tf.float32)
+
+    # Linear activation, using rnn inner loop last output
+    return tf.matmul(states[1], weights['out']) + biases['out']
+
+logits = RNN(X, weights, biases)
+prediction = tf.nn.softmax(logits)
+
+# Define loss and optimizer
+cost = tf.reduce_mean(tf.nn.softmax_cross_entropy_with_logits(
+    logits=logits, labels=Y))
+train_op = tf.train.AdamOptimizer(learning_rate).minimize(cost)
+
+# Evaluate model (with test logits, for dropout to be disabled)
+correct_pred = tf.equal(tf.argmax(prediction, 1), tf.argmax(Y, 1))
+accuracy = tf.reduce_mean(tf.cast(correct_pred, tf.float32))
+
+# Initialize the variables (i.e. assign their default value)
+init = tf.global_variables_initializer()
+
+# Start training
+with tf.Session() as sess:
+
+    # Run the initializer
+    sess.run(init)
+
+    for step in range(1, training_steps+1):
```

Comment:
import `xrange` from `six` and use that.

Reply:
Plesae use pylint

Reply:
it seems range is more recommended than xrange.